### DISCUSSION TITLE
Lost one row, after rolling window

### DISCUSSION THREAD

**Discussion by b-y-f** [blue-yonder/tsfresh#1012] [Q&A]
```python
import pandas as pd
import numpy as np

date_rng = pd.date_range(start='1/1/2020', end='1/2/2020', freq='30min')
df = pd.DataFrame(date_rng, columns=['timestamp'])
df['value'] = np.random.randint(0,100,size=(len(date_rng)))
df['meter_id'] = np.random.randint(1,2,size=(len(date_rng)))
df
```

Above code will generate 49 rows of data, after I do some feature extraction for window, I got 48 rows of data
```python3
def preprocess_for_tsfresh(df: pd.DataFrame):
    # Assuming your dataframe is named df
    df["id"] = df["meter_id"].astype(str) + "_" + df["timestamp"].astype(str)

    df_shift, y = make_forecasting_frame(
        df["value"], kind="meter", max_timeshift=10, rolling_direction=1
    )
    return df_shift
```

```
preprocess_for_tsfresh(df).groupby("id").count() # 48 rows 1id to id=48
```

**Comment by nils-braun** [comment]
That is not a big but intentional: make forecasting frame prepares your data for a forecasting training task. In forecasting, you want to predict the next value from the previous features. As we do not have the next data point for the last row, we can not use it for training and will therefore drop it. 
If you want more control I would use the more general roll_time_series function as described in the forecasting notebook.
